ballerina/tests/client_endpoint_test.bal

```diff
@@ -566,112 +477,68 @@ public function testDeleteFile() returns error? {
 }
 
 @test:Config{
     dependsOn: [testRemoveDirectoryWithFiles]
 }
-public function testRemoveComplexDirectory() {
-    Error? response1 = clientEp->rmdir("/home/in/complexDirectory");
-    if response1 is Error {
-        test:assertFail(msg = "Error while invoking the `rmdir` operation" + response1.message());
+public function testRemoveComplexDirectory() returns error? {
+    return testGenericRmdir("/home/in/complexDirectory");
+}
+
+function testGenericRmdir(string path) returns error? {
+    boolean|Error response0 = clientEp->isDirectory(path);
+    log:printInfo("Executed `isDirectory` operation before deleting a directory " + path);
+    int retryCount = 0;
+    while ((response0 is Error || !response0) && retryCount < 10) {
+        log:printInfo("Executed `isDirectory` operation before deleting a directory " + path);
+        runtime:sleep(1);
+        response0 = clientEp->isDirectory(path);
+        retryCount += 1;
+    }
+
+    if retryCount >= 10 {
+        test:assertFail(msg = "Error while invoking the `isDirectory` operation before invoking the `rmdir` operation");
     } else {
-        log:printInfo("Executed `rmdir` operation");
+        Error? response1 = clientEp->rmdir(path);
+        if response1 is Error {
+            test:assertFail(msg = "Error while invoking the `rmdir` operation on " + path + ": " + response1.message());
+        } else {
+            log:printInfo("Executed `rmdir` operation on " + path);
+        }
     }
 
-    boolean|Error response2 = clientEp->isDirectory("/home/in/complexDirectory");
-    log:printInfo("Executed `isDirectory` operation after deleting a directory");
+    boolean|Error response2 = clientEp->isDirectory(path);
+    log:printInfo("Executed `isDirectory` operation after deleting a directory " + path);
+
     int i = 0;
-    while (response2 is boolean && response2 && i < 10) {
-         runtime:sleep(1);
-         response2 = clientEp->isDirectory("/home/in/complexDirectory");
-         log:printInfo("Executed `isDirectory` operation after deleting a directory");
-         i += 1;
+    while (response2 is boolean && i < 10) {
+        runtime:sleep(1);
+        response2 = clientEp->isDirectory(path);
+        log:printInfo("Executed `isDirectory` operation after deleting a directory " + path);
+        i += 1;
     }
     if response2 is Error {
-        test:assertEquals(response2.message(), "/home/in/complexDirectory does not exists to check if it is a directory.",
+        test:assertEquals(response2.message(), path + " does not exists to check if it is a directory.",
             msg = "Incorrect error message for non-existing file/directory at `isDirectory` operation after `rmdir` operation");
     } else {
-        // test:assertFail(msg = "Error not created while invoking `isDirectory` operation after `rmdir` operation" );
+        // test:assertFail(msg = "Error not created while invoking `isDirectory` operation after `rmdir` operation on " + path );
```

Comment:
Shouldn't we uncomment this line? otherwise, the test will pass in both conditions?

Reply:
There is commented due to https://github.com/ballerina-platform/ballerina-standard-library/issues/1448